pkg/asset/manifests/vsphere/cloudproviderconfig.go

```diff
@@ -26,13 +28,18 @@ func CloudProviderConfig(folderPath string, p *vspheretypes.Platform) (string, e
 	fmt.Fprintln(buf, "")
 
 	fmt.Fprintln(buf, "[Workspace]")
-	printIfNotEmpty(buf, "server", p.VCenter)
+	host, port, err := net.SplitHostPort(p.VCenter)
+	if err != nil {
+		host = p.VCenter
+	}
+	printIfNotEmpty(buf, "port", port)
```

Comment:
we should not be setting the port when the vCenter didn't have that in the first place right? otherwise we would set it to the zero value (0)

Reply:
printIfNotEmpty will only set if there is a non empty string in port value. Port is a string so ""(empty string) will not be considered.